instructions/07-querying-flink-tables.md

```diff
@@ -0,0 +1,240 @@
+# Querying Flink Tables with the Table API (Exercise) 
+
+In this exercise, we will execute some Flink Table API queries using the built-in Marketplace tables in Confluent Cloud.
+
+We'll start with basic select statements to set a foundation for future exercises.
+
+## Stage the exercise
+
+Stage the exercise by executing:
+
+```
+./exercise.sh stage 07
+```
+
+You should now see a `CustomerService.java` and `OrderService.java` in your `marketplace` package and a series of tests for each.
+
+You can run the tests from your IDE, or by executing the following Maven command:
+
+```
+mvn clean test
+```
+
+Currently, these tests will fail because the methods haven't been implemented.
+
+While working through the exercise, it can be helpful to periodically run the tests to measure progress.
+
+## Query all customers
+
+The first task is to implement the `CustomerService.allCustomers` method.
+
+This is an easy-to-understand query. We are trying to obtain the details for all of the customers.
+
+Most Flink applications will modify the data before sending it to another table. Here we use the data as is and don't send it anywhere. This means it is of limited use.
+
+However, it isn't useless. These queries can often be useful when debugging an application to understand what data is coming in.
+
+More importantly, it creates a foundation for future work.
+
+**Note:** It's important to remember that the data in the table is streaming and unbounded. Once the query is executed it will run forever until it is terminated.
+
+### Implement the method
+
+Implement the `CustomerService.allCustomers` method as follows:
+
+- Select all fields from all customers in the `examples.marketplace.customers` table.
+- Execute and return the result.
+
+<details>
+  <summary>**Hint**</summary>
+  <p>
+You can implement a basic select statement as follows:
+	
+```
+env.from("TABLE NAME")
```

Comment:
I like how Hints are just that. No solutions, just hints.
This made me think - would it make sense to add links to the actual solutions? Or they'll have it all cloned to their machines anyway, so it's not necessary?

Reply:
During the first exercise, they will be reading through the README which explains how they can pull the solution if necessary.

Links to the solution are an interesting idea though. Let me think on that.